tests/posix/common/src/nanosleep.c

```diff
@@ -186,22 +199,19 @@ static void common(const uint32_t s, uint32_t ns)
 	zassert_equal(rem.tv_nsec, 0, "actual: %d expected: %d",
 		rem.tv_nsec, 0);
 
-	ns += s * NSEC_PER_SEC;
-
-	dt = now - then;
-	dt_ns = k_cyc_to_ns_ceil64(dt);
-	if (dt_ns == 0) {
-		/* k_cycle_get_32() does not seem to be completely accurate on
-		 * some virtual platforms in CI (some function calls to
-		 * nanosleep reportedly take 0ns).
-		 */
-		dt_ns = 1;
-	}
-
-	printk("dt_ns: %llu\n", dt_ns);
-
-	zassert_true(dt_ns >= ns, "expected dt_ns >= %d: actual: %d", (int)ns,
-		(int)dt_ns);
+	uint64_t exp_ns = s * NSEC_PER_SEC + ns;
+	uint64_t tolerance_ns = MAX(NSEC_PER_CYCLE,
```

Comment:
what is the reason for capping it with NSEC_PER_CYCLE?

Reply:
It's not being capped: it's being cupped (? whatever the opposite is).  I.e. the minimum value is NSEC_PER_CYCLE, necessary to handle the cases where `exp_ns` is so small that even with tolerance it's a sub-cycle amount.  A one-cycle minimum is necessary because the ceiling is used when converting cycles to ns.

This test is supposed to work on 1ns delays, which should round up to a one cycle delay due to timer resolution.